qp_qiime2/tests/test_qiime2.py

```diff
@@ -41,9 +41,10 @@ def tearDown(self):
                     remove(fp)
 
     def test_rarefy(self):
-        params = {'p-sampling-depth': 2, 'i-table': 5}
+        params = {'Sampling depth': 2, 'BIOM table': 5}
```

Comment:
`5` doesn't look like a biom table or path to a biom table. Would it be more correct to describe this as "BIOM table artifact ID"?

Reply:
I think the blending of UI and compute logic is confusing personally.

Reply:
IMO, the way I see this, is that the plugin is doing the translation between the compute logic and the UI. At the end, the actual logic is in Q2 not here...

Reply:
It seems like the responsibilities of this plugin are to:

* perform computational work directly using the qiime2 API
* interact with the qiita database for parameters, artifacts, etc
* ensure the above fulfills requirements for a singular and logically distinct user interface

Is that accurate?

Reply:
Close but not exactly.
The plugin is the interface between Qiita and the underlying tool (in this case Qiime2).

It is not using the Qiime2 API, but the Qiime2 CLI (@antgonza may have more information about this decision).

It doesn't interact with the Database directly, it uses the Qiita REST api to retrieve the different values that it needs.

Since Qiita doesn't know anything about the plugin, the plugin itself needs to make sure to tell Qiita how he wants the different parts of the plugin to be shown to the user.